scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -502,66 +702,104 @@ harmony_pcs <- harmony::HarmonyMatrix(
 
+Let's highlight the cell types in the harmony UMAP:
 
-```{r harmony umap celltypes}
+```{r harmony umap celltypes broad}
 scater::plotReducedDim(merged_sce,
-                       # plot the harmony coordinates
                        dimred = "harmony_UMAP",
                        # color by broad celltypes
                        colour_by = "celltype_broad",
-                       # Some styling to help us see the points:
                        point_size = 0.5,
-                       point_alpha = 0.2, 
+                       point_alpha = 0.2) +
+  ggtitle("UMAP after integration with harmony") +
+  guides(colour = guide_legend(override.aes = list(size = 3, alpha = 1))) 
+```
+Similar to the `fastMNN` results, the non-tumor cell types do appear to have been well-integrated across batches, but that one library does stand out!
+We can see that 
+
+Now is a good time to remember we actually have two cell type variables: `celltype_broad` and `celltype_fine`, where the latter contains finer-grained Tumor characterizations based on differentiation stage.
+Let's see what we observe if we color the UMAP instead by the finer-grained cell types:
+
+```{r harmony umap celltypes fine}
+scater::plotReducedDim(merged_sce,
+                       dimred = "harmony_UMAP",
+                       # color by FINE celltypes
+                       colour_by = "celltype_fine",
+                       point_size = 0.5,
+                       point_alpha = 0.1, 
                        other_fields = "sample") +
-  ggplot2::ggtitle("UMAP after integration with harmony") +
-  # Modify the ggplot2 guides so that the legend points are larger and easier to see:
-  ggplot2::guides(colour = ggplot2::guide_legend(override.aes = list(size = 3, alpha = 1))) +
-  ggplot2::facet_wrap(~sample, nrow=2)
+  ggtitle("UMAP after integration with harmony") +
+  guides(colour = guide_legend(override.aes = list(size = 3, alpha = 1))) +
+  facet_wrap(~sample)
+  
+```
+
+It looks as though `SCPCL000481` may indeed be comprised of relatively distinct cell types, which we can check with a barplot:
+
+<!-- LATE BREAKING ADDITION IN NEED OF REVIEW/DISCUSSION, including where to put this plot? Should I make this plot? -->
+```{r count celltypes fine}
+# Create a table of counts sample and celltype_fine combinations
+sample_celltype_table <- colData(merged_sce) %>%
+  as.data.frame() %>%
+  dplyr::count(sample, celltype_fine) 
+
+
+sample_celltype_table
+
+# Let's look at just tumor cells:
+sample_celltype_table %>%
+  dplyr::filter(stringr::str_detect(celltype_fine, "Tumor")) %>%
+  ggplot(aes(x = sample, y = n, fill = celltype_fine)) +
+  geom_col(position = "dodge")
 ```
 
-How do you think these methods' (`fastMNN` and `harmony`) performances compare on these data?
+Indeed, we see that `SCPCL000481` has a substantial `Tumor_Myocyte-A` population that the other libraries do not have, and the two libraries from the same patient show similar trends. 
```

Comment:
Although it has a population of `Tumor_Myocyte-A` that other libraries don't have it still has overlapping cell types... so it should have some overlapping cells with other samples. You see that with `fastMNN`, but not with `Harmony`. I do think this is helpful in showing why some parts of the samples may or may not be integrating, but I'm not sure how in detail we want to go. Curious what other people think.

Reply:
I am not sure we need to go into this level of detail... By the time we get here, everyone will probably be pretty tired. 

We can point out the `Tumor_Myocyte-A` population, but I agree with Ally that it does seem like the other overlapping populations should allow for integration!